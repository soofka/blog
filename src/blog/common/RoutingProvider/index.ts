import * as queryString from 'query-string';

import {
  homeNiceUrlBasePath,
  aboutMeNiceUrlBasePath,
  talksNiceUrlBasePath,
  archiveNiceUrlBasePath,
  entryNiceUrlBasePath,
  entriesByTagNiceUrlBasePath,
  entriesByDateNiceUrlBasePath,
  errorNiceUrlBasePath,
} from './constants';

export class RoutingProvider {
  static getHomeNiceUrlBasePath(): string {
    return homeNiceUrlBasePath;
  }

  static getAboutMeNiceUrlBasePath(): string {
    return aboutMeNiceUrlBasePath;
  }

  static getTalksNiceUrlBasePath(): string {
    return talksNiceUrlBasePath;
  }

  static getArchiveNiceUrlBasePath(): string {
    return archiveNiceUrlBasePath;
  }

  static getEntryNiceUrlBasePath(): string {
    return entryNiceUrlBasePath;
  }

  static getEntriesByTagNiceUrlBasePath(): string {
    return entriesByTagNiceUrlBasePath;
  }

  static getEntriesByDateNiceUrlBasePath(): string {
    return entriesByDateNiceUrlBasePath;
  }

  static getErrorNiceUrlBasePath(): string {
    return errorNiceUrlBasePath;
  }

  static setInQueryString(object: any) {
    const oldQueryStringObject = queryString.parse(location.search);
    const newQueryStringObject = { ...oldQueryStringObject, ...object };
    history.pushState(null, '', `?${queryString.stringify(newQueryStringObject)}`);
  }

  static parseTextToNiceUrl(text: string): string {
    return text
      .trim()
      .toLowerCase()
      .replace(/\s+/g, '-')
      .replace(/[!|?|.|,|:|;|(|)|{|}|\[|\]|\\|\/]/g, '');
  }
}

export default RoutingProvider;
