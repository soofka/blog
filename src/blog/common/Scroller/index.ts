export class Scroller {
  static scrollToTop(): void {
    Scroller.scrollTo({ top: 0 });
  }

  static scrollToEntryTop(): void {
    Scroller.scrollTo({ top: 144 });
  }

  static scrollTo(position: { top?: number }): void {
    window.scrollTo({
      ...position,
      behavior: 'smooth',
    });
  }
}

export default Scroller;
