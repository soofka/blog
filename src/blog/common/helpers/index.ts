export * from './assets';
export * from './cookie';
export * from './date';
export * from './labels';
export * from './language';
export * from './queryString';
export * from './request';
export * from './scroll';
export * from './webWorker';
