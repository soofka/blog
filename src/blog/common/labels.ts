export const labels = {
  en: {
    page_title: 'soofka.pl',
    page_title__about_me: 'About me - soofka.pl',
    page_title__talks: 'Talks - soofka.pl',
    page_title__archive: 'Archive - soofka.pl',
    page_title__entry: '%entry_title% - soofka.pl',
    page_title__entries_by_tag: '%tag% entries - soofka.pl',
    page_title__entries_by_date: 'entries from %date% - soofka.pl',
    page_description: `
      Personal blog of Jakub Sowiński. Focused on programming, mainly full stack web development.
      %description%
    `,
    page_keywords: 'html, css, php, javascript, react, %keywords%',
    menu__header: 'My blog',
    menu_item__home: 'Home',
    menu_item__about: 'About me',
    menu_item__talks: 'Talks',
    menu_item__archive: 'Archive',
    errors__header: 'Error!',
    errors__default: 'Something went wrong. Please try again.',
    errors__page_not_found: 'Whoops, there is no such page.',
    errors__no_entry_content: 'Whoops, file with content of this entry crashed. Please try again.',
    errors__no_data_in_db: 'Whoops, database crashed. Please try again.',
    errors__no_entries_matching_query: 'Sorry, there are no entries matching this query.',
    entry__back: 'Back to list of entries',
    entry__created: 'created on ',
    entry__last_updated: ', last updated on ',
    entry__continue_reading: 'Continue reading',
    quote: `
      Passionate of programming in general, lately focused on JavaScript particularly.
      Ultimately, I would like to have some major contribution to development of our civilization,
      still thinking how JS can help me with that task though. Also interested in astronomy, board games, and e-sports.
      `,
    scroll_to_top: 'Scroll to top',
  },
  pl: {
    page_title: 'soofka.pl',
    page_title__about_me: 'O mnie - soofka.pl',
    page_title__talks: 'Wystąpienia - soofka.pl',
    page_title__archive: 'Archiwum - soofka.pl',
    page_title__entry: '%entry_title% - soofka.pl',
    page_title__entries_by_tag: 'wpisy o %tag% - soofka.pl',
    page_title__entries_by_date: 'wpisy z %date% - soofka.pl',
    page_description: `
      Prywatny blog Jakuba Sowińskiego. Piszę o programowaniu, głównie full stack web developmencie.
      %description%
    `,
    page_keywords: 'html, css, php, javascript, react, %keywords%',
    menu__header: 'Mój blog',
    menu_item__home: 'Home',
    menu_item__about: 'O mnie',
    menu_item__talks: 'Wystąpienia',
    menu_item__archive: 'Archiwum',
    errors__header: 'Błąd!',
    errors__default: 'Coś poszło nie tak. Spróbuj ponownie.',
    errors__page_not_found: 'Ups, nie ma takiej strony.',
    errors__no_entry_content: 'Ups, plik z treścią tego wpisu się popsuł. Spróbuj ponownie.',
    errors__no_data_in_db: 'Ups, baza się popsuła. Spróbuj ponownie.',
    errors__no_entries_matching_query: 'Brak wpisów pasujących do tego zapytania.',
    entry__back: 'Wróć do listy wpisów',
    entry__created: 'utworzony ',
    entry__last_updated: ', ostatnio zmodyfikowany ',
    entry__continue_reading: 'Czytaj dalej',
    quote: `
      Zafascynowany programowaniem w ogóle, ostatnimi czasy skoncentrowany na JavaScript w szczególności.
      Ostatecznie chciałbym wnieść znaczący wkład w rozwój naszej cywilizacji, wciąż jednak zastanawiam się
      jak w tym celu może pomóc mi JS. Interesuję się również astronomią, grami planszowymi oraz e-sportem.
    `,
    scroll_to_top: 'Wróć na górę',
  },
};
