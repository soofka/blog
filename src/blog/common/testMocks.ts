export const mockStoreProps = {
  languageStore: {
    getLanguage: () => 'en',
    setLanguage: () => null,
  },
};

export const mockRouteProps = {
  match: {
    params: {
      tag: 'testTag',
      date: 'testDate',
      niceUrl: 'testNiceUrl',
    },
  },
};
