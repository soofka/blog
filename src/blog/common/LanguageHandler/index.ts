import * as queryString from 'query-string';
import * as cookies from 'js-cookie';

import RoutingProvider from 'common/RoutingProvider';

import {
  languages,
  defaultLanguage,
  languageParameter,
} from './constants';

class LanguageHandler {

  static getCurrentLanguage(): string {
    const queryStringParsed = queryString.parse(location.search);

    if (queryStringParsed && queryStringParsed[languageParameter]) {
      const languageFromQueryString = queryStringParsed[languageParameter].toLowerCase();

      if (this.isLanguageValid(languageFromQueryString)) {
        return languageFromQueryString;
      }
    }

    const languageFromCookie = cookies.get('language');
    if (languageFromCookie && this.isLanguageValid(languageFromCookie)) {
      return languageFromCookie;
    }

    return defaultLanguage;
  }

  static setCurrentLanguage(language: string): void {
    cookies.set('language', language);
    RoutingProvider.setInQueryString({ [languageParameter]: language });
  }

  static isLanguageValid(language: string): boolean {
    for (const key in languages) {
      if (languages.hasOwnProperty(key)) {
        if (language === languages[key]) {
          return true;
        }
      }
    }
    return false;
  }
}

export default LanguageHandler;
