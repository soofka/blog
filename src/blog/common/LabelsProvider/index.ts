import labels from './labels';
import { defaultLanguage } from 'common/LanguageHandler/constants';

export class LabelsProvider {
  static getLabel(key: string, params: any = {}, language: string = defaultLanguage): string {
    const label = labels[language][key];

    if (label) {
      return LabelsProvider.replaceInLabel(labels[language][key], params);
    }

    console.log(`Label key ${key} in language ${language} not found.`);
    return key;
  }

  static replaceInLabel(label: string, params: any): string {
    if (params && Object.keys(params).length) {
      let newLabel = label;

      for (const key in params) {
        if (params.hasOwnProperty(key)) {
          newLabel = newLabel.replace(`%${key}%`, params[key]);
        }
      }

      return newLabel;
    }

    return label;
  }
}

export default LabelsProvider;
