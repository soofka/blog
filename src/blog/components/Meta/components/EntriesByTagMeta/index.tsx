import { inject, observer } from 'mobx-react';
import * as React from 'react';

import { getLabel } from 'blog/common/helpers';
import { LanguageStoreInterface } from 'blog/store/language';

import { Meta } from '../../';

interface EntriesByTagMetaPropsInterface {
  tag: string;
  languageStore?: LanguageStoreInterface;
}

export const EntriesByTagMetaComponent = (props: EntriesByTagMetaPropsInterface) => {
  const { tag, languageStore: { getLanguage } } = props;
  return <Meta
    title={getLabel('page_title__entries_by_tag', { tag }, getLanguage())}
    keywords={getLabel('page_keywords', { keywords: tag }, getLanguage())}
  />;
};

export const EntriesByTagMeta = inject('languageStore')(observer(EntriesByTagMetaComponent));
