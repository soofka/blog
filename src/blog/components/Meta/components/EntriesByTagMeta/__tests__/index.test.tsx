import * as React from 'react';
import { shallow } from 'enzyme';

import { mockStoreProps } from 'common/testMocks';

import EntriesByTagMeta from 'components/Meta/components/EntriesByTagMeta';

describe('<EntriesByTagMeta/>', () => {

  it('renders correctly', () => {
    expect(shallow(
      <EntriesByTagMeta
        tag="testTag"
        {...mockStoreProps}
      />,
    )).toMatchSnapshot();
  });

});
