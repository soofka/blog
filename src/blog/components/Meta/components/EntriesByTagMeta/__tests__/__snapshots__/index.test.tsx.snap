// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntriesByTagMeta/> renders correctly 1`] = `
<_class
  languageStore={
    Object {
      "getLanguage": [Function],
      "setLanguage": [Function],
    }
  }
  tag="testTag"
/>
`;
